adaptive_scheduler/kernel/slicedipscheduler_v2.py

```diff
@@ -93,7 +93,13 @@ def build_data_structures(self):
                     scheduled = 1
                 # now w_idx is the index into r.possible_starts, which have
                 # been reordered by time.
-                self.Yik.append([r.resID, w_idx, r.priority, ps.resource, scheduled])
+                if r.request and r.request.optimization_type == 'AIRMASS':
+                    # Apply the airmass coefficient into the priority
+                    priority = r.priority + ps.airmass_coefficient
+                else:
+                    # Add the earlier window optimization priority factor to the effective priority
+                    priority = r.priority + (0.1 / (w_idx + 1.0))
+                self.Yik.append([r.resID, w_idx, priority, ps.resource, scheduled])
```

Comment:
This is the important part where the priority gets perturbed based on the optimization type